node/src/components/chain_synchronizer/error.rs

```diff
@@ -47,6 +47,9 @@ pub(crate) enum Error {
         activation_point: EraId,
     },
 
+    #[error("no blocks have been found in storage (should have at least genesis switch block)")]
```

Comment:
```suggestion
    #[error("no blocks have been found in storage (should provide recent trusted hash)")]
```

Reply:
This (well, not the message, but this error) actually causes problem for `reactor::participating::tests::run_participating_network` - one of the nodes in that test gets started after the genesis timestamp, but before the first era ends. IIRC, the expectation so far was that you can join without a trusted hash as long as the genesis era hasn't ended, but now, because we don't create the genesis switch block after genesis timestamp has passed, the node can't join because it _needs_ at least one trusted block, and it has none.

Should we change the criterion for creating the genesis block to "if the current timestamp isn't later than minimum era duration after genesis"?

Reply:
The test has been modified to allow all nodes to start before genesis in [ac72aa3](https://github.com/casper-network/casper-node/pull/3280/commits/ac72aa39d469d02a2103b1e2818009588049b33d)